hw/battery/src/battery.c

```diff
@@ -487,10 +487,15 @@ battery_set_poll_rate_ms(struct os_dev *battery, uint32_t poll_rate)
 {
     struct battery *bat = (struct battery *)battery;
 
-    if ((poll_rate == 0) || (bat == NULL)) {
+    if ((bat == NULL)) {
         return -1;
     }
 
+    if (poll_rate == 0) {
+        os_callout_stop(&battery_manager.bm_poll_callout);
+        return 0;
```

Comment:
This is another minor thing: I think it would be better to set `bat->b_poll_rate` to 0 before returning.  That way, the contents of the data structure reflect the actual state of the system.  If someone is debugging live or looking at a core, a non-zero value here could be misleading.